src/xrpld/ledger/detail/View.cpp

```diff
@@ -278,6 +309,46 @@ accountHolds(
         view, account, issue.currency, issue.account, zeroIfFrozen, j);
 }
 
+STAmount
+accountHolds(
+    ReadView const& view,
+    AccountID const& account,
+    MPTIssue const& mptIssue,
+    FreezeHandling zeroIfFrozen,
+    AuthHandling zeroIfUnauthorized,
+    beast::Journal j)
+{
+    STAmount amount;
+
+    auto const sleMpt =
+        view.read(keylet::mptoken(mptIssue.getMptID(), account));
+    if (!sleMpt)
+        amount.clear(mptIssue);
+    else if (
+        zeroIfFrozen == fhZERO_IF_FROZEN && isFrozen(view, account, mptIssue))
+        amount.clear(mptIssue);
+    else
+    {
+        amount = STAmount{mptIssue, sleMpt->getFieldU64(sfMPTAmount)};
+
+        // only if auth check is needed, as it needs to do an additional read
+        // operation
+        if (zeroIfUnauthorized == ahZERO_IF_UNAUTHORIZED)
```

Comment:
i believe it will be inevitably used in V2. Although ahZERO_IF_UNAUTHORIZED is not used,  I still think we should keep enum to make things clear in the code how we evaluate the MPT balance in specific usages

Reply:
I'm wondering if we should treat this like the Locked Amount from earlier, and remove it until it is actually used. If it never ends up being used, we don't have this useless extra check lying around.